### ISSUE TITLE
Cannot override config of extensions appended in techdocs-core

### ISSUE THREAD

**Issue by kunickiaj** [backstage/mkdocs-techdocs-core#220]
For example, if one wants to enable slugify on content tabs, they cant because the extension is always appended even if already configured in the user's mkdocs.yml

I saw https://github.com/backstage/mkdocs-techdocs-core/pull/214 which i think intended to address this issue, but i still cant get configuration to get merged with the default atechdocs one.

https://github.com/backstage/mkdocs-techdocs-core/blob/8bc1e2a79dc56f160458f6a8ab65393277a01456/src/core.py#L127

Have also tried

```yaml
plugins:
  - techdocs-core
markdown_extensions:
  - pymdownx.extra:
      pymdownx.tabbed:
        slugify: !!python/object/apply:pymdownx.slugs.slugify
          kwds:
            case: lower
```

I suspect the issue is that while #214 removed the superfences plugin from the list of appended, it did not remove the others provided by `pymdownx.extra` still resulting in impossible configuration for those.

**Comment by kunickiaj** [comment]
After some more investigation I see that the config is correctly applied in normal mkdocs serve (on 8080) even with the techdocs-core plugin added so it seems something is being stripped out (dont know if this depends on some js) but something in techdocs is preventing the generation of the slugified anchor links.

**Comment by deejay1** [comment]
@kunickiaj which version of `techdocs-cli` are you using? this was only recently fixed and should be in 1.31.1

**Comment by kunickiaj** [comment]
Using techdocs-core 1.4.2 and backstage 1.30.1 and whatever the latest CLI is I think.

I did try to patch around this but still had no success in the slugify tabs to work.
